src/test/java/aist/edge/edgeservice/EdgeServiceIntegrationTests.java

```diff
@@ -0,0 +1,104 @@
+package aist.edge.edgeservice;
+
+import static org.assertj.core.api.Java6Assertions.assertThat;
+
+import com.palantir.docker.compose.DockerComposeRule;
+import com.palantir.docker.compose.connection.waiting.HealthChecks;
+
+import java.util.*;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.http.*;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.util.*;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringBootTest(classes = EdgeServiceApplication.class, webEnvironment = SpringBootTest.WebEnvironment.DEFINED_PORT)
+public class EdgeServiceIntegrationTests {
+
+    @ClassRule
+    public static DockerComposeRule docker = DockerComposeRule.builder().pullOnStartup(true)
+            .file("src/test/resources/docker-compose.yml")
+            .waitingForService("microservice--user-service", HealthChecks.toHaveAllPortsOpen())
+            .waitingForService("microservice--user-service", HealthChecks.toRespondOverHttp(8091,
+                (port) -> port.inFormat("http://localhost:8091")))
```

Comment:
In the future we should tie these port numbers in with the configuration service, making only the config service port the only hardcoded one.